src/CheckFor.php

```diff
@@ -210,4 +210,98 @@ public static function isType($desired_type, $row) {
     }
     return TRUE;
   }
+
+  /**
+   * Check file for redirects.
+   *
+   * @param object $row
+   *   A row object as delivered by migrate.
+   *
+   * @return bool
+   *   TRUE - full URI(s) of the redirect destination(s).
+   *   FALSE - no detectable redirects exist in the page.
+   */
+  public static function hasHtmlRedirect($row) {
+    // Checks for meta location redirects.
+    $meta_location = qp($row)->find('meta[http-equiv="location"')->attr("url");
+    if (!empty($meta_location)) {
+      return $meta_location;
+    }
+    // Checks for meta refresh redirects. Does support relative urls.
+    $meta_refresh = qp($row)->find('meta[http-equiv="refresh"]')->attr("url");
+    if (!empty($meta_refresh)) {
+      return $meta_refresh;
+    }
+    // Checks for presence of Javascript. <script type="text/javascript">
+    $js = qp($row)->find('script[type="text/javascript"]');
+    if (!empty($js)) {
+      // Checks for window location JS redirects. window.location.href
+      $wlh = $js->attr('window.location.href');
+      if (!empty($wlh)) {
+        return $wlh;
+      }
+      // Checks for location JS redirects. location
+      $location = $js->attr('location');
+      if (!empty($location)) {
+        return $location;
+      }
+      // Checks for location href JS redirects. location.href
+      $lh = $js->attr('location.href');
+      if (!empty($lh)) {
+        return $lh;
+      }
+      // Checks for location assign JS redirects. location.assign
+      // Checks for location JS redirects. location
+      $la = $js->attr('location.assign');
+      if (!empty($la)) {
+        return $la;
+      }
+      // Checks for location replace JS redirects. location.replace
+      // Checks for location JS redirects. location
+      $lr = $js->attr('location.replace');
+      if (!empty($lr)) {
+        return $lr;
+      }
+    }
+
+    // Checks for human readable text redirects.
+    $human = qp($row)->find(body);
+      // Checks for 'this page has been moved to'.
+      $has_been = $human->text('This page has been moved to');
+      if (!empty($has_been)) {
+        return $has_been;
+      }
+      // Checks for 'this page has moved to'.
+      $this_page = $human->text('this page has moved to');
+      if (!empty($this_page)) {
+        return $this_page;
+      }
+  }
+
+  /**
+   * @param $redirect_url
```

Comment:
Missing first line of doc block.

This method looks great. Though it is not specific to redirects. It has other potential uses so I would make the param just be $url.


This method is probably better placed in Url.php.

The hasHtmlRedirect method I think is fine in CheckFor because it is used for checking the entire page for something, not for manipulating an url.